src/test/shell/integration/configured_query_test.sh

```diff
@@ -1008,24 +1023,31 @@ def format(target):
     return str(target.label) + '%None'
   first = str(bo['//command_line_option:platform_suffix'])
   second = str(('//$pkg:myflag' in bo) and bo['//$pkg:myflag'])
-  return str(target.label) + '%' + first + '%' + second
+  third = str(bo['//$pkg:mylistflag'] if '//$pkg:mylistflag' in bo else None)
+  return str(target.label) + '%' + first + '%' + second + '%' + third
 EOF
 
   bazel cquery "//$pkg:bar" --output=starlark \
     --starlark:file=$pkg/expr.star > output 2>"$TEST_log" || fail "Expected success"
 
-  assert_contains "//$pkg:bar%None%False" output
+  assert_contains "//$pkg:bar%None%False%None" output
```

Comment:
Why is this `None` instead of the build_setting_default of `["c"]`?

Reply:
This is a consequence of how Starlark options are implemented: When they haven't been set anywhere, they aren't tracked and the dict of all options just won't contain any value for them, not even the default.

This makes sense given the decentralized nature of Bazel: Knowing all Starlark options and their default values would require loading all packages (and thus all repos) eagerly.

Reply:
That's why `build_options` is "magic": It gives you access to the full internal view of all options that Bazel maintains. This has an entry for `mylistflag` simply because it has been set on the command-line. `build_options` doesn't care about dependencies here, which is why it is only made available in `cquery`, not in regular Starlark.